safekeeper/src/send_wal.rs

```diff
@@ -479,6 +481,9 @@ impl SafekeeperPostgresHandler {
             // todo: add read|write .context to these errors
             r = sender.run() => r,
             r = reply_reader.run() => r,
+            _ = tli_cancel.cancelled() => {
```

Comment:
That's true -- I think it makes sense to prefer to avoid this top level drop-futures-on-cancel behavior, as in future I could imagine WAL sending code becoming cancellation-unsafe (e.g. if in future we wrote control files from this path, or if we downloaded segments from S3 onto local disk instead of streaming them).  In f176b95a1902f02646e2312efca4561e46b5012e I've removed this check.

Reply:
For the record: Erik persuaded me that we're better off doing the top-level cancellation and trusting that the code is cancel-safe rather than pre-emptively doing fine-grained cancellation (c8623b55ff6b5ff051a5f8f2d3c08e8a12015a1b)